platform-api/src/main/java/io/datacater/core/deployment/K8Deployment.java

```diff
@@ -318,24 +323,85 @@ private List<EnvVar> getEnvironmentVariables(
+    streamOutConfig.remove(StaticConfig.VALUE_SERIALIZER);
+
+    /**
+     * Set `group.id` to the UUID of the deployment, such that deployments with more than one
+     * replica can make use of Apache Kafka's consumer group protocol.
+     */
+    streamInConfig.put(StaticConfig.GROUP_ID, uuid);
 
-    try {
-      variables.add(
+    List<EnvVar> environmentVariables = new ArrayList<>();
+
+    // Pass serde config as environment variables
+    for (String configOption : dataCaterConfig.keySet()) {
+      String configOptionAsEnvVariable = transformConfigOptionToEnvVariable(configOption);
+      environmentVariables.add(
           createEnvVariable(
-              StaticConfig.DC_STREAMIN_CONFIG_TEXT,
-              objectMapper.writeValueAsString(streamInConfig)));
-      variables.add(
+              StaticConfig.DATACATER_SERDE_ENV_PREFIX,
+              configOptionAsEnvVariable,
+              dataCaterConfig.get(configOption).toString()));
+    }
+
+    // Pass stream in config as environment variables
+    for (String configOption : streamInConfig.keySet()) {
+      String configOptionAsEnvVariable = transformConfigOptionToEnvVariable(configOption);
+      environmentVariables.add(
+          createEnvVariable(
+              StaticConfig.STREAMIN_ENV_PREFIX,
+              configOptionAsEnvVariable,
+              streamInConfig.get(configOption).toString()));
+    }
+
+    // Pass stream out config as environment variables
+    for (String configOption : streamOutConfig.keySet()) {
+      String configOptionAsEnvVariable = transformConfigOptionToEnvVariable(configOption);
+      environmentVariables.add(
           createEnvVariable(
-              StaticConfig.DC_STREAMOUT_CONFIG_TEXT,
-              objectMapper.writeValueAsString(streamOutConfig)));
-    } catch (JsonProcessingException e) {
-      throw new CreateDeploymentException(StringUtilities.wrapString(e.getMessage()));
+              StaticConfig.STREAMOUT_ENV_PREFIX,
+              configOptionAsEnvVariable,
+              streamOutConfig.get(configOption).toString()));
     }
 
-    return variables;
+    // Set topic name of stream in and stream out
+    environmentVariables.add(
+        createEnvVariable(StaticConfig.STREAMIN_ENV_PREFIX, "TOPIC", streamIn.getName()));
+    environmentVariables.add(
+        createEnvVariable(StaticConfig.STREAMOUT_ENV_PREFIX, "TOPIC", streamOut.getName()));
+
+    return environmentVariables;
+  }
+
+  /**
+   * Transforms a configuration name to an environment variable.
+   *
+   * <p>Examples: - `datacater.name` becomes `DATACATER_NAME` - `datacater.middle-level.name`
+   * becomes `DATACATER_MIDDLE_LEVEL_NAME`
+   *
+   * @param configName
+   * @return config name as valid environment variable
+   */
+  private String transformConfigOptionToEnvVariable(String configName) {
+    return configName
+        // Replace all characters, except a-z A-Z and underscores, with an underscore
+        .replaceAll("[^a-zA-Z_]", "_")
```

Comment:
Yes, this would help a lot. I am also not a fan of it.

AFAIK, the Kafka consumer and producer APIs do not use any properties with a `dash` in the name, so this should not cause any harm.

Nonetheless I still prefer to keep this part for making sure that we do not pass any invalid notation to the environment variables. I am aware of libraries that use it, e.g., [Smallrye](https://quarkus.io/guides/kafka#configuring-smallrye-kafka-connector).